boot/freeldr/install/linux/install.c

```diff
@@ -0,0 +1,144 @@
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include <ctype.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdbool.h> 
+#include "install.h"
+#include "volume.h"
+
+/*
+ * These includes are required to define
+ * the fat_data and fat32_data arrays.
+ */
+#include "fat.h"
+#include "fat32.h"
+
+bool BackupBootSector(char* lpszVolumeName);
+bool InstallBootSector(char* lpszVolumeType);
+int stricmp(const char *a, const char *b);
+
+int main(int argc, char *argv[])
+{
+    if (argc < 3)
+    {
+        printf("syntax: install x: [fs_type]\nwhere fs_type is fat or fat32\n");
+        return -1;
+    }
+
+    if (!OpenVolume(argv[1]))
+    {
+        printf("Exiting program...\n");
+        return -1;
+    }
+    
+	if (!InstallBootSector(argv[2]))
+    {
+        printf("Exiting program...\n");
+        return -1;
+    }
+    
+    CloseVolume();
+
+    printf("You must now copy freeldr.sys & freeldr.ini to %s.\n", argv[1]);
+
+    return 0;
+}
+
+
+bool InstallBootSector(char* lpszVolumeType)
+{
+    unsigned char BootSectorBuffer[512];
+
+    if (!ReadVolumeSector(0, BootSectorBuffer))
+    {
+        return false;
+    }
+
+    if (stricmp(lpszVolumeType, "fat") == 0)
+    {
+        //
+        // Update the BPB in the new boot sector
+        //
+        memcpy((fat_data+3), (BootSectorBuffer+3), 59 /*fat BPB length*/);
+
+        //
+        // Write out new boot sector
+        //
+        if (!WriteVolumeSector(0, fat_data))
+        {
+            printf("Exiting program...\n");
+            return false;
+        }
+    }
+    else if (stricmp(lpszVolumeType, "fat32") == 0)
+    {
+        //
+        // Update the BPB in the new boot sector
+        //
+        memcpy((fat32_data+3), (BootSectorBuffer+3), 87 /*fat32 BPB length*/);
+
+        //
+        // Write out new boot sector
+        //
+        if (!WriteVolumeSector(0, fat32_data))
+        {
+            printf("Exiting program...\n");
+            return false;
+        }
+
+        //
+        // Write out new extra sector
+        //
+        if (!WriteVolumeSector(14, (fat32_data+512)))
+        {
+            printf("Exiting program...\n");
+            return false;
+        }
+    }
+    else
+    {
+        printf("%s:%d: ", __FILE__, __LINE__);
+        printf("File system type %s unknown.\n", lpszVolumeType);
+        return false;
+    }
+
+    printf("%s boot sector installed.\n", lpszVolumeType);
+
+    return true;
+}
+
+int stricmp(const char *a, const char *b)
```

Comment:
This function is already available.

Reply:
Not really, it is non-standard.

Reply:
Then you have the choice of doing something similar to what is already done for the host-tools, namely:
```c
#ifdef _WIN32
#include <string.h>         // Required for _stricmp()
#else
#include <strings.h>        // Required for strcasecmp()
#define _stricmp strcasecmp
#endif
```

Reply:
OK, didn't knew that.